sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPlugin.scala

```diff
@@ -28,9 +33,45 @@ import org.scalajs.jsenv.jsdomnodejs.JSDOMNodeJSEnv
 import org.scalajs.jsenv.phantomjs.PhantomJSEnv
 
 object ScalaJSPlugin extends AutoPlugin {
-  /** The global Scala.js IR cache */
+  /** The global Scala.js IR cache.
+   *
+   *  If you want to allocate a new cache for your purposes, prefer calling
+   *  [[newIRCache]] rather than
+   *  [[org.scalajs.core.tools.io.IRFileCache.newCache globalIRCache.newCache]].
+   *  The former will automatically register the new cache to be freed when the
+   *  build is unloaded. If you use the latter, it is your responsibility to
+   *  free the cache.
+   */
   val globalIRCache: IRFileCache = new IRFileCache()
```

Comment:
We should consider hiding this. Do we know of any use case that needs it? Easier to open than close later.

Reply:
I don't know any use case, at the moment. Except that it appears in the path dependent type of `scalaJSIRCache`, which is a `TaskKey[globalIRCache.Cache]`. In 0.6.x that has to stay, but in 1.x we could potentially use `IRFileCache#Cache` instead.

Reply:
FTR, neither GitHub nor Google can find `globalIRCache` anywhere but in our own codebase.

Reply:
In that case, I propose we deprecate it all in 0.6.x and hide it entirely in 1.0.0 unless someone complains. This also solves the problem that it would be one of the few places where the sbt plugin explicitly exposes something from `linker.standard._` (we haven't yet published a version with this, right?).
We can put a comment on `scalaJSIR` that it really is preferable because it is cached :)

Reply:
> (we haven't yet published a version with this, right?).

That's right.

We can make it private in 1.0.0, but we can't deprecate it in 0.6.x for the same reason as above: it is part of the path-dependent type of something that is not deprecated. We can hide it back in `ScalaJSPluginInternal` in 0.6.x, though, since it was still there in the latest released version.

Reply:
> This also solves the problem that it would be one of the few places where the sbt plugin explicitly exposes something from linker.standard._

`IRFileCache` is in `org.scalajs.core.tools.io`, not in `standard._`. It has to be, since it is virtually necessary to create the correct type of input to `Linker.link`.

Reply:
Right... Shouldn't review code when I'm too tired :P

So I guess we hide it back into internal now and make it private in 1.0.0. I'd really rather just have `scalaJSIR` for people to use and expose this only if someone needs it.

Reply:
OK